### PR TITLE
ForwardDiff in tests only

### PR DESCRIPTION
**Pull request by rgiordan**
This moves all the ForwardDiff tests into a single file, adds a command line flag to runtests, and uses analytic gradients and Hessians for the KL divergence, which removes ForwardDiff from the actual inference.

Many of the tests currently sequestered to test_derivatives.jl are important.  I'd like to merge this now before finding a more permanent solution to avoid more merges as everyone else continues to make changes.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -4,7 +4,7 @@ DataFrames
 Distributions
 DocOpt
 Dtree
-ForwardDiff 0.1.8 0.2
+ForwardDiff 0.2
 JLD
 Optim 0.5 0.5.1
```
Comment (jeff-regier) [inline REQUIRE:9]: Can we remove the upper bound from Optim now that ForwardDiff is free?

File: doc/celeste_equations.lyx [hunk 2]
```diff
@@ -1630,6 +1630,170 @@ Also,
```
Comment (jeff-regier) [inline doc/celeste_equations.lyx:1630]: Why are the equations changing? (Is it somehow related to ForwardDiff?)

File: src/Model.jl [hunk 3]
```diff
@@ -30,7 +30,6 @@ import Base.+
 import Distributions
 import FITSIO
 import WCS.WCSTransform
-import ForwardDiff
```
Comment (jeff-regier) [inline src/Model.jl:33]: Nice! Glad to have this out of here.

**Comment by jeff-regier** [comment]
Looks good Ryan. 

I think it'd be good practice to start squashing commits into one, or into logical groups, before committing to master, i.e. `git rebase HEAD~13` in this case. There's a post about it here:
https://programmers.stackexchange.com/questions/263164/why-squash-git-commits-for-pull-requests

Main idea is to keep the history readable.

File: doc/celeste_equations.lyx (hunk 2 above)
Comment (jeff-regier) [inline doc/celeste_equations.lyx:1630]: I think with `git rebase`, you can reorder commits as well as squash related commits into one. If these doc changes are different from the ForwardDiff change, you might squash the ForwardDiff changes together, but leave this one as its own.

File: REQUIRE (hunk 1 above)
Comment (rgiordan) [inline REQUIRE:9]: I think so, but I'm just going to fix the Optim stuff in a separate CL.

File: doc/celeste_equations.lyx (hunk 2 above)
Comment (rgiordan) [inline doc/celeste_equations.lyx:1630]: I added the formulas for the KL gradients and Hessians.

**Comment by rgiordan** [comment]
I'll close this and re-open a PR that's been rebased.

**Comment by jeff-regier** [comment]
Actually I've found that of just rebasing the branch automatically updates
the pr. GitHub is pretty slick...I figured that changing the history would
throw it off

**Pull request closed without merging**